### PR TITLE
Cleanup: use klog.Info in pkg/volume/util/operationexecutor/operation_generator.go

### LINKED ISSUE #103907: Operationexecutor log format problem
#### What happened:
`I0726 15:05:12.072600   39595 operation_generator.go:974] UnmountDevice succeeded for volume "volume-name" %!(EXTRA string=UnmountDevice succeeded for volume "volume-name" (UniqueName: "fake-plugin/fake-device1") on node "mynodename" )
`

Here at the same time, simple, detail information is returned and formatting problems are caused.
#### What you expected to happen:
No format error. For example, only datail is returned.
#### How to reproduce it (as minimally and precisely as possible):
```shell
cd pkg/kubelet/volumemanager/reconciler
go test . -v
```
#### Anything else we need to know?:
Found from running pkg/kubelet/volumemanager/reconciler/reconciler_test.go

#### Environment:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

/good-first-issue

### PR DESCRIPTION
**Pull request by devinleee**
Signed-off-by: yuswift <yuswift2018@gmail.com>

#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/operationexecutor/operation_generator.go#L974
prints too many logs. I think using `GenerateMsgDetailed` is enough.
#### Which issue(s) this PR fixes:

Fixes #103907 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by yuswift** [comment]
/cc @njuptlzf @yuzhiquan 
Could you take a look at this pr? Thanks.

**Comment by njuptlzf** [comment]
Thank you @yuswift for this pr.

Structured log migration is being done #101760, 
if #101760 is needed, the work here will move there, 
Can you clarify whether #101760 is needed? @serathius 
otherwise lgtm.

**Comment by njuptlzf** [comment]
If the #101760 wait period is very long, I think it is necessary here, then #101760 use `rebase`. @serathius

**Review by yangjunmyfm192085 — COMMENTED** [review]
/lgtm
/assign @jingxu97

/triage accepted
/sig instrumentation
/wg structured-logging

**Review by yangjunmyfm192085 — COMMENTED** [review]
/retest

**Review by saschagrunert — APPROVED** [review]

**Comment by yuswift** [comment]
@saschagrunert could you add an `approve` label to this pr?

**Comment by saschagrunert** [comment]
> @saschagrunert could you add an `approve` label to this pr?

I don't think that I have the power, only @saad-ali and @jingxu97 can.

**Comment by dims** [comment]
this is waiting on @saad-ali and/or @jingxu97. thanks!

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 1]
```diff
@@ -971,7 +971,7 @@ func (og *operationGenerator) GenerateUnmountDeviceFunc(
 			return volumetypes.NewOperationContext(eventErr, detailedErr, migrated)
 		}
 
-		klog.Infof(deviceToDetach.GenerateMsg("UnmountDevice succeeded", ""))
+		klog.Infof(deviceToDetach.GenerateMsgDetailed("UnmountDevice succeeded", ""))
```
Comment (pohly) [inline pkg/volume/util/operationexecutor/operation_generator.go:974]: While at it, please replace with `klog.Info`. `Infof` interprets the first parameter as a format string, which wasn't (and still isn't) what gets passed here.

**Review by pohly — CHANGES_REQUESTED** [review]
/cancel lgtm

File: pkg/volume/util/operationexecutor/operation_generator.go (hunk 1 above)
Comment (pohly) [reply pkg/volume/util/operationexecutor/operation_generator.go:974]: @swiftslee: please address this comment.

Comment (swiftslee) [reply pkg/volume/util/operationexecutor/operation_generator.go]: > @swiftslee: please address this comment.

sorry for the miss. done

**Review by pohly — APPROVED** [review]
/lgtm
/assign @xing-yang 

For approval.

@swiftslee For the future: take a look at https://www.kubernetes.dev/docs/guide/pull-requests/#commit-message-guidelines

As it stands now, "fix log format problem" doesn't tell a reviewer what the problem is and why it needs to be fixed.

**Comment by swiftslee** [comment]
> /lgtm /assign @xing-yang
> 
> For approval.
> 
> @swiftslee For the future: take a look at https://www.kubernetes.dev/docs/guide/pull-requests/#commit-message-guidelines
> 
> As it stands now, "fix log format problem" doesn't tell a reviewer what the problem is and why it needs to be fixed.

Thanks for your advice. done

**Comment by pohly** [comment]
> Thanks for your advice. done

As you changed the commit message, let's discuss this a bit further :grin: 

What your commit message contains now is not the "why" for this change, but the "what":  `Cleanup: use klog.Info in pkg/volume/util/operationexecutor/operation_generator.go`

The real question that needs to be answered is: "why is Info correct here and Infof isn't"? The "what" can be determined easily by looking at the diff (which file, which code change in that file), but the "why" is often harder (not necessarily here, but the principle is the same). It's not uncommon for people to look at code again a few years later and then ask around "does anybody remember why this code is like this" - for example, see https://github.com/kubernetes/kubernetes/pull/108902#discussion_r833549204

A better commit message would be something like this:
```
pkg/volume: fix incorrect klog.Infof usage

klog.Infof expects a format string as first parameter and then
expands format specifies inside it. What gets passed here
is the final string that must be logged as-is, therefore
klog.Info has to be used.
```

**Comment by swiftslee** [comment]
> > Thanks for your advice. done
> 
> As you changed the commit message, let's discuss this a bit further 😁
> 
> What your commit message contains now is not the "why" for this change, but the "what": `Cleanup: use klog.Info in pkg/volume/util/operationexecutor/operation_generator.go`
> 
> The real question that needs to be answered is: "why is Info correct here and Infof isn't"? The "what" can be determined easily by looking at the diff (which file, which code change in that file), but the "why" is often harder (not necessarily here, but the principle is the same). It's not uncommon for people to look at code again a few years later and then ask around "does anybody remember why this code is like this" - for example, see [#108902 (comment)](https://github.com/kubernetes/kubernetes/pull/108902#discussion_r833549204)
> 
> A better commit message would be something like this:
> 
> ```
> pkg/volume: fix incorrect klog.Infof usage
> 
> klog.Infof expects a format string as first parameter and then
> expands format specifies inside it. What gets passed here
> is the final string that must be logged as-is, therefore
> klog.Info has to be used.
> ```



> > Thanks for your advice. done
> 
> As you changed the commit message, let's discuss this a bit further 😁
> 
> What your commit message contains now is not the "why" for this change, but the "what": `Cleanup: use klog.Info in pkg/volume/util/operationexecutor/operation_generator.go`
> 
> The real question that needs to be answered is: "why is Info correct here and Infof isn't"? The "what" can be determined easily by looking at the diff (which file, which code change in that file), but the "why" is often harder (not necessarily here, but the principle is the same). It's not uncommon for people to look at code again a few years later and then ask around "does anybody remember why this code is like this" - for example, see [#108902 (comment)](https://github.com/kubernetes/kubernetes/pull/108902#discussion_r833549204)
> 
> A better commit message would be something like this:
> 
> ```
> pkg/volume: fix incorrect klog.Infof usage
> 
> klog.Infof expects a format string as first parameter and then
> expands format specifies inside it. What gets passed here
> is the final string that must be logged as-is, therefore
> klog.Info has to be used.
> ```

Really learned a lot! Please take a look at the latest commit message. BTW, the e2e test failed. Should we ignore it or not?

**Review by pohly — APPROVED** [review]
/lgtm

**Comment by pohly** [comment]
> Please take a look at the latest commit message.

Kubernetes doesn't use "signed-off-by", but it doesn't hurt either.

> BTW, the e2e test failed. Should we ignore it or not?

We can in this case because the failures are clearly not related to the change in this PR. The unit test failures are known flakes, too.

/test pull-kubernetes-unit

**Comment by swiftslee** [comment]
> > Please take a look at the latest commit message.
> 
> Kubernetes doesn't use "signed-off-by", but it doesn't hurt either.
> 
> > BTW, the e2e test failed. Should we ignore it or not?
> 
> We can in this case because the failures are clearly not related to the change in this PR. The unit test failures are known flakes, too.
> 
> /test pull-kubernetes-unit

Unit test passed.

**Pull request merged**